src/widgets/moonraker/component.jsx

```diff
@@ -0,0 +1,51 @@
+import { useTranslation } from "next-i18next";
+
+import Container from "components/services/widget/container";
+import Block from "components/services/widget/block";
+import useWidgetAPI from "utils/proxy/use-widget-api";
+
+export default function Component({ service }) {
+  const { t } = useTranslation();
+
+  const { widget } = service;
+
+  const { data: printStats, error: printStatsError } = useWidgetAPI(widget, "print_stats");
+  const { data: displayStatus, error: displayStatsError } = useWidgetAPI(widget, "display_status");
+  const { data: webHooks, error: webHooksError } = useWidgetAPI(widget, "webhooks");
+
+  if (printStatsError || displayStatsError || webHooksError) {
+    const finalError = printStatsError ?? displayStatsError ?? webHooksError;
+    return <Container error={finalError} />;
+  }
+
+  if (!printStats || !displayStatus || !webHooks) {
+    return (
+      <Container service={service}>
+        <Block label="moonraker.printer_state" />
+      </Container>
+    );
+  }
+
+  if (webHooks.result.status.webhooks.state === "shutdown") {
+    return (
+      <Container service={service}>
+        <Block label="moonraker.printer_state" value={webHooks.result.status.webhooks.state} />
+      </Container>
+    );
+  }
+
+  let currentLayer = "-";
+  let totalLayer = "-";
+  if(!printStats.result.status.print_stats.info.total_layer === "") {
```

Comment:
```suggestion
  if (printStats.result.status.print_stats.info.total_layer !== "") {
```

I think should be this, at the moment youre (unintentionally) coercing the value to a boolean which doesnt really work. Depending on the type of the API output though there could be another way

Reply:
Yeah, i think you are right. TBH i don't know what my brain was thinking while i wrote that.
```suggestion
  if(printStats.result.status.print_stats.info.total_layer !== null) {
```
This should be even better since I am using strict equality and the API returns null instead of 0 if it has no layer information.